Userland/Services/WebContent/WebDriverConnection.cpp

```diff
@@ -1807,6 +1813,43 @@ ErrorOr<void, Web::WebDriver::Error> WebDriverConnection::handle_any_user_prompt
     return {};
 }
 
+// https://w3c.github.io/webdriver/#dfn-waiting-for-the-navigation-to-complete
+ErrorOr<void, Web::WebDriver::Error> WebDriverConnection::wait_for_navigation_to_complete()
+{
+    // 1. If the current session has a page loading strategy of none, return success with data null.
+    if (m_page_load_strategy == Web::WebDriver::PageLoadStrategy::None)
+        return {};
+
+    // 2. If the current browsing context is no longer open, return success with data null.
+    TRY(ensure_open_top_level_browsing_context());
```

Comment:
This isn't right - this will throw an error, but we should return success.
```c++
if (m_page_client.page().top_level_browsing_context().has_been_discarded())
    return {};
```